server/src/test/java/org/elasticsearch/ExceptionsHelperTests.java

```diff
@@ -221,4 +221,265 @@ public void testCauseCycle() {
         ExceptionsHelper.unwrap(e1, IOException.class);
         ExceptionsHelper.unwrapCorruption(e1);
     }
+
+    public void testLimitedStackTrace() {
```

Comment:
These tests look really good already. Maybe the cherry on top would be one test where the stacktrace has fewer items than the max?